specification.md

```diff
@@ -259,6 +259,39 @@ customary in the language in which the library is implemented
 
 ### JobExecutor
 
+The `JobExecutor` represents one or more concrete mechanisms for
+executing jobs. It contains all the operations that are specific to a
+particular such mechanism. Specifically, it knows how to start a job
+through the `submit()` call, query the status of jobs and inform client
+API of any updates through callbacks, and can `cancel()` a running job.
+
+Client code interacts with a concrete job execution mechanism by invoking
+methods on objects declared (in a strictly-typed language) as
+`JobExecutor`. This leaves a number of possible ways to structure an
+implementation of  this API:
```

Comment:
Is the intention behind exposing multiple ways to structure this class to support different languages, with each language only using one possible way?  Or is the intention to have both ways supported within the same implementation?

My main reason for asking is out of source-code compatibility concerns.

Reply:
I think it's because, as far as I remember, we didn't reach a consensus on one or the other. Or, rather said, each choice may be more appropriate for a specific scenario, such as implementing the API from scratch or writing a wrapper for an existing library.

I wrote a bit on this in #72, but there are other places where discussions around the exact nature of `JobExecutor` happened.

Reply:
>>    LGTM. I echo @SteVwonder's point about source code portability.

> So do I. I can either remove that part or we can clarify once we reach a consensus.

I would suggest to add a statement along the lines of:

```
Language bindings SHOULD explicitly define an mechanism to ensure source code compatibility between implementations.
```

Reply:
I added a note to this extent. Please let me know if that works. If not I can also replace with the above suggested statement.

Reply:
Works for me, thanks

Reply:
Works for me as well. Thanks!